ada/language/ast.py

```diff
@@ -8751,12 +8773,8 @@ def return_type():
                     Entity.info.from_rebound
                 ),
 
-                entity_no_md(
-                    BaseTypeDecl,
-                    Entity.info.md.primitive_real_type.cast(BaseTypeDecl),
-                    Entity.info.rebindings,
-                    Entity.info.from_rebound
-                ),
```

Comment:
Would be nice to have a small comment explaining why this new logic is the right one

Reply:
What if I have no idea ?

Reply:
We're good to go then :+1: